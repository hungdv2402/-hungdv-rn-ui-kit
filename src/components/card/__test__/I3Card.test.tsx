import renderer from 'react-test-renderer';
import I3Card from '../I3Card';
import React from 'react';
import { Text } from 'react-native';

describe('renders I3Card correctly', () => {
  it('should render I3Card with padding={16} round={8} color="lightpink"', () => {
    const tree = renderer
      .create(
        <I3Card
          padding={16}
          round={8}
          color="lightpink"
        >
          <Text>should render I3Card with padding={16} round={8} color="lightpink"</Text>
        </I3Card>
      ).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('should render I3Card with custom style', () => {
    const tree = renderer
      .create(
        <I3Card style={{ marginHorizontal: 16 }} >
          <Text>should render I3Card with custom style</Text>
        </I3Card>
      ).toJSON();
    expect(tree).toMatchSnapshot();
  });
});