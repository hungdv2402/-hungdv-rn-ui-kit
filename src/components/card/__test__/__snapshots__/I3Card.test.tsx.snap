// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders I3Card correctly should render I3Card with custom style 1`] = `
<View
  style={
    [
      {
        "marginHorizontal": 16,
      },
      {
        "backgroundColor": "#fff",
        "borderRadius": 16,
        "padding": 16,
      },
    ]
  }
>
  <Text>
    should render I3Card with custom style
  </Text>
</View>
`;

exports[`renders I3Card correctly should render I3Card with padding={16} round={8} color="lightpink" 1`] = `
<View
  style={
    [
      undefined,
      {
        "backgroundColor": "lightpink",
        "borderRadius": 8,
        "padding": 16,
      },
    ]
  }
>
  <Text>
    should render I3Card with padding=
    16
     round=
    8
     color="lightpink"
  </Text>
</View>
`;
